apollo/interfaces/generic/platforms/aws_generic/platform.py

```diff
@@ -0,0 +1,85 @@
+import os
+from typing import Dict, Optional
+
+import logging
+
+from apollo.agent.constants import PLATFORM_AWS_GENERIC
+from apollo.agent.env_vars import (
+    STORAGE_BUCKET_NAME_ENV_VAR,
+    GUNICORN_WORKERS_ENV_VAR,
+    GUNICORN_THREADS_ENV_VAR,
+    GUNICORN_TIMEOUT_ENV_VAR,
+    MCD_AGENT_CLOUD_PLATFORM_ENV_VAR,
+)
+from apollo.agent.models import AgentConfigurationError
+from apollo.interfaces.lambda_function.platform import AwsPlatformProvider
+
+logger = logging.getLogger(__name__)
+
+
+class AwsGenericPlatformProvider(AwsPlatformProvider):
+    """
+    AWS generic platform provider
+    """
+
+    def __init__(self, platform_info: Optional[Dict] = None, **kwargs):  # type: ignore
+        super().__init__(**kwargs)
+        self._platform_info = platform_info or {
+            STORAGE_BUCKET_NAME_ENV_VAR: os.getenv(STORAGE_BUCKET_NAME_ENV_VAR),
+            GUNICORN_WORKERS_ENV_VAR: os.getenv(GUNICORN_WORKERS_ENV_VAR),
+            GUNICORN_THREADS_ENV_VAR: os.getenv(GUNICORN_THREADS_ENV_VAR),
+            GUNICORN_TIMEOUT_ENV_VAR: os.getenv(GUNICORN_TIMEOUT_ENV_VAR),
+            MCD_AGENT_CLOUD_PLATFORM_ENV_VAR: os.getenv(
+                MCD_AGENT_CLOUD_PLATFORM_ENV_VAR
+            ),
+        }
+
+    @property
+    def platform_info(self) -> Dict:
+        return self._platform_info
+
+    @property
+    def platform(self) -> str:
+        return PLATFORM_AWS_GENERIC
+
+    @property
+    def updater(self) -> None:
+        # automatic updates are not supported for AWS generic deployments
+        return None
+
+    def get_infra_details(self) -> Dict:
+        return {}
+
+    def filter_log_events(
```

Comment:
I think I need to do a bit more refactoring to expose these functions to the agent's API

Reply:
I think you're right - I'm about to commit a refactoring to do that

Reply:
refactored. now lambda + aws_generic both use the same `main.py`